zcash_client_sqlite/src/lib.rs

```diff
@@ -717,24 +710,80 @@ impl<P: consensus::Parameters> WalletWrite for WalletDb<rusqlite::Connection, P>
                 }
             }
 
+            #[cfg(feature = "orchard")]
+            #[allow(unused_assignments)] // Remove this when the todo!()s below are implemented.
+            for output in d_tx.orchard_outputs() {
+                match output.transfer_type() {
+                    TransferType::Outgoing | TransferType::WalletInternal => {
+                        let recipient = if output.transfer_type() == TransferType::Outgoing {
+                            // TODO: Recover the actual UA, if possible.
+                            Recipient::Unified(
+                                UnifiedAddress::from_receivers(
+                                    Some(output.note().recipient()),
+                                    None,
+                                    None
+                                ).expect("UA has an Orchard receiver by construction."),
+                                PoolType::Shielded(ShieldedProtocol::Orchard)
+                            )
+                        } else {
+                            Recipient::InternalAccount(
+                                *output.account(),
+                                Note::Orchard(*output.note()),
+                            )
+                        };
+
+                        wallet::put_sent_output(
+                            wdb.conn.0,
+                            &wdb.params,
+                            *output.account(),
+                            tx_ref,
+                            output.index(),
+                            &recipient,
+                            output.note_value(),
+                            Some(output.memo()),
+                        )?;
+
+                        if matches!(recipient, Recipient::InternalAccount(_, _)) {
+                            todo!();
+                            //wallet::orchard::put_received_note(wdb.conn.0, output, tx_ref, None)?;
+                        }
+                    }
+                    TransferType::Incoming => {
+                        match spending_account_id {
+                            Some(id) => {
+                                if id != *output.account() {
+                                    panic!("Unable to determine a unique account identifier for z->t spend.");
```

Comment:
This isn't a z->t spend at least because we're in orchard (not sapling).
I also don't see anything suggesting this is an unshielding transaction. We're just looping over orchard outputs here.
Why is the local variable called `spending_account_id` when in fact it is set to the value of the _receiving_ account id?
And why must every output in a transaction have the same receiving account? This looks like a panic will occur for any transaction that sends outputs to more than one account tracked by the sqlite db. 
And in fact, I found this code due to a panic when doing account recovery for a testnet account I have, and a multi-account transaction is very likely to be what I have here (though I haven't confirmed this.
See also [discord post](https://discord.com/channels/809218587167293450/876655911790321684/1219086198513860659).